omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/UserResource2_0.java

```diff
@@ -49,7 +49,9 @@ public UserAndPassword1_8 save(UserAndPassword1_8 user) {
 		} else {
 			openmrsUser = Context.getUserService().saveUser(user.getUser());
 			Context.refreshAuthenticatedUser();
-			if (openmrsUser.getId() != null && StringUtils.isNotBlank(user.getPassword())) {
+			if (openmrsUser.getId() != null && (Context.getAuthenticatedUser().getUuid().equals(openmrsUser.getUuid()))) {
```

Comment:
I would expect the logic to be

    if (Context.hasPrivilege(PrivilegeConstants.EDIT_USER_PASSWORDS)) {
        // call the admin version
    } else {
        // call the changeOwnPassword function
    }